sql/catalyst/src/test/scala/org/apache/spark/sql/types/StructTypeSuite.scala

```diff
@@ -606,4 +612,181 @@ class StructTypeSuite extends SparkFunSuite with SQLHelper {
         "b STRING NOT NULL,c STRING COMMENT 'nullable comment'")
     assert(fromDDL(struct.toDDL) === struct)
   }
+
+  test("simple struct with collations to json") {
+    val simpleStruct = StructType(
+      StructField("c1", StringType(UNICODE_COLLATION)) :: Nil)
+
+    val expectedJson =
+      s"""
+         |{
+         |  "type": "struct",
+         |  "fields": [
+         |    {
+         |      "name": "c1",
```

Comment:
Let's also have tests with names with dots. Also what about tests looking at case sensitivity? And special characters?

Reply:
there are already tests with dots in `DataTypeSuite` because we care about deserialization there as well so that's why I didn't put them here. I also added there tests for case sensitivity and special characters.